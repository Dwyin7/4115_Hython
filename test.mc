import x;

def int test(int a, float b, char c){
    int x; 
    x = a + b + c;
    return x;
}

def int test2(int x, float y, char z){
    int a; 
    INT b = [1,2];
    INT B = [[1,2,3],[4,5,6]];
    INT A = [[],[],[],[]];
    int c = 12;
    a = x + y + z;
    return a;
}




#
## this is a comment
#
#
#int a = 1;
#int b = 2;
#int c = 3;
#
#test(a,b,c);