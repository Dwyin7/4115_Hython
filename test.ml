open Ast

let _ =
  let input_string = input_line stdin in
  print_endline "Input from stdin:";
  print_endline input_string;
  let lexbuf = Lexing.from_string input_string in
  let program = Parser.program_rule Scanner.token lexbuf in
  print_endline "\nParsed program:";
  print_endline (string_of_program program)