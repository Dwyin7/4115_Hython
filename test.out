

Parsed program: 

Import -> lib: x

Function -> Output Type: P_int, Name: test2, Formals: [(P_int x), (P_float y), (P_char z)],
Body: [
Expr (Bind(P_int, a));
Expr (Bind(P_int, b));
Expr (BindAndAssign (P_int c, Assign exprs Int(5)));
Expr (Assign(a, Binop(Id(x), +, Binop(Id(y), +, Id(z)))));
Return (Id(a))
]
Function -> Output Type: P_int, Name: test, Formals: [(P_int a), (P_float b), (P_char c)],
Body: [
Expr (Bind(P_int, x));
Expr (Assign(x, Binop(Id(a), +, Binop(Id(b), +, Id(c)))));
Return (Id(x))
]


