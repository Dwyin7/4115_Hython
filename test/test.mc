import x;

def int test_primitive(int a, float b, char c){
    int x; 
    x=1;
    y=[[1,3,4,5],[1,3,4,5],[1,3,4,5],[1,3,4,5]];
    z = [1,3,4,5];
    while(x<=100){
        x = x + 1;
    }
    for(x in y){
        int i = x;
    }

    return x;
}

def int test_matrix(int x, float y, char z){
    int a; 
    INT b = [1,2];
    INT B = [[1,2,3],[4,5,6]];
    INT A = [[],[],[],[]];
    int c = 12;
    a = x + y + z;
    return a;
}


x = test_primitive(1,2,3);

#
## this is a comment
#
#
#int a = 1;
#int b = 2;
#int c = 3;
#
#test(a,b,c);