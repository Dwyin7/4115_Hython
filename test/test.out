from x import 

int test_primitive(int a, float b, char c) {
int x;
x = 1;
y = Tensor(Tensor(1, 3, 4, 5), Tensor(1, 3, 4, 5), Tensor(1, 3, 4, 5), Tensor(1, 3, 4, 5));
z = Tensor(1, 3, 4, 5);
while ((x <= 100)) {x = (x + 1);}
for (x in y) {int i = x;}
return x;
}
int test_matrix(int x, float y, char z) {
int a;
T_int b = Tensor(1, 2);
T_int B = Tensor(Tensor(1, 2, 3), Tensor(4, 5, 6));
T_int A = Tensor(Tensor(), Tensor(), Tensor(), Tensor());
int c = 12;
a = ((x + y) + z);
return a;
}
x = test_primitive(1, 2, 3);
